rmgpy/data/thermo.py

```diff
@@ -2020,7 +2020,7 @@ def compute_group_additivity_thermo(self, molecule):
         # Generate estimates of the thermodynamics parameters
         for atom in molecule.atoms:
             # Iterate over heavy (non-hydrogen) atoms
-            if atom.is_non_hydrogen():
+            if atom.is_non_hydrogen() and not atom.is_halogen():
```

Comment:
Is this temporary or permanently the right thing? H atoms seem to be treated specially below around line 2034; should halogens be treated similarly?   Is H special because it's terminal, one-short-of-valent (is there a term for that?) or because it's light, or something else?   
If this is all correct, and we're right to just be iterating over any non-terminal atoms, or non-valence-7 atoms (rather than "heavy" atoms) then please just update the comment on line 2022 which says "heavy (non-hydrogen)".  Or if this is pending future work (eg. more groups) open a new issue so we don't forget.

Reply:
This is temporary, until we added hypervalent chlorine, bromine and iodine groups (which we don't currently have atom types for yet).  Currently, halogens are terminal, and therefore the thermo is accounted for by central atom (like CF3 group in CF3-R).  We have dummy halogen groups in the thermo database so that, when we descend the tree with a halogen, it hits the dummy node with zeros.  I added `not atom.is_halogen()` here to avoid descending the tree at all, because it will be zeros.  However, since we will want to descend the tree if/when we add hypervalent groups, perhaps I should remove this since the dummy halogen groups are already there to account for terminal halogens.

Reply:
The comment was updated and makes more sense. resolving conversation.